src/module/config.c

```diff
@@ -3,3 +3,8 @@ unsigned int krf_personality = 0x001c;
 unsigned int krf_probability = 1000;
 unsigned int krf_targeted_uid = 1002;
 unsigned int krf_log_faults = 0;
+unsigned int krf_pid_target = 0;
```

Comment:
This is a solid first approach, but I think we can structure this better for future extensibility. Instead of continuing to add individual variables + an option mask, we could have a structure in *config.h*. A very rough sketch:

```c
#define KRF_T_MODE_MAX 31
#define KRF_T_MODE_MAX_MASK (1 << KRF_T_MODE_MAX)

typedef enum {
  KRF_T_MODE_PERSONALITY = (1 << 0);
  KRF_T_MODE_PID = (1 << 1);
  // ...
} krf_target_mode_mask_t;

typedef struct {
  krf_target_mode_mask_t mode_mask;
  unsigned int target_data[KRF_T_MODE_MAX];
} krf_target_options_t;

extern krf_target_options_t krf_target_options;
```

And rewrite the `KRF_TARGETED` macro into an (inlinable?) function that tests `krf_target_options` instead.

Similarly for the `procfs` sources/outputs: we can just have a single file that reads `>=2` `unsigned int`s: the first is the mask, followed by each mask's parameter in ascending-bit order. Setting the mask to `0` disables all targeting and flushes any parameters from `target_data`.

Reply:
We could also make target configuration way simpler: instead of allowing the user to configure `target_data` in one shot, we could require them to write to the `procfs` file once for each mask. That's what they currently have to do for configuring syscalls, and that's a reasonable approach.

Reply:
There's some problem with:
> unsigned int target_data[KRF_T_MODE_MAX];


```c
14:15 <disconnect3d> << (1<<31)
14:15 <cxx> -2147483648
14:16 <disconnect3d> unsigned int t[1<<31];
14:16 <cxx> error: narrowing conversion of '-2147483648' from 'int' to 'long unsigned int'
```

Reply:
`KRF_T_MODE_MAX` would just be `31`, so it should just be `unsigned int[31]`. It's `KRF_T_MODE_MAX_MASK` that would be `1 << 31` 😉

Reply:
Lol, my bad :D kk :)